indico/modules/events/registration/fields/simple.py

```diff
@@ -318,18 +389,30 @@ def view_data(self):
         for item in items:
             item['caption'] = self.form_item.data['captions'][item['id']]
         data['choices'] = items
+        data['places_used'] = self.get_places_used()
         return data
 
     @property
     def validators(self):
         def stay_dates_valid(form, field):
             data = snakify_keys(field.data)
-            if _to_date(data['arrival_date']) > _to_date(data['departure_date']):
+            if data and (_to_date(data['arrival_date']) > _to_date(data['departure_date'])):
                 raise ValidationError(_("Arrival date can't be set after the departure date."))
-        return [stay_dates_valid]
+
+        def _check_number_of_places(form, field):
+            if field.data:
+                item = next((x for x in self.form_item.versioned_data['choices'] if x['id'] == field.data['choice']),
+                            None)
+                captions = self.form_item.data['captions']
+                if item and item['places_limit'] and not ((item['places_limit']
+                                                           - self.get_places_used()[field.data['choice']])):
+                    raise ValidationError(_('Not enough rooms in the {0}').format(captions[item['id']]))
+        return [stay_dates_valid, _check_number_of_places]
 
     def get_friendly_data(self, registration_data):
         friendly_data = dict(registration_data.data)
+        if not friendly_data:
```

Comment:
I'd do that before. `if not registration_data.data: return {}` - no need to copy the empty dict first